packages/flutter/lib/src/widgets/scroll_activity.dart

```diff
@@ -671,3 +671,45 @@ class DrivenScrollActivity extends ScrollActivity {
     return '${describeIdentity(this)}($_controller)';
   }
 }
+
+//// An activity that a scroll view based on delta
```

Comment:
IIRC there was logic in the scroll system that would cancel scroll activities when starting new ones; am I misremembering? If that's the case, won't a steady stream of small scroll events cause a bunch of cancellations of previous scroll events?

Reply:
I added log to observe and did not find what you are concerned about, scheduleMicrotask will execute before the next event arrives.

But this part needs to be redesigned because of the overscroll issue, and perhaps not using ScrollActivity would be a better option to avoid the overscroll issue with applyUserOffset.